verifiers/internal/gcb/provenance.go

```diff
@@ -0,0 +1,374 @@
+package gcb
+
+import (
+	"crypto/sha256"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"os"
+	"regexp"
+	"strings"
+
+	intoto "github.com/in-toto/in-toto-golang/in_toto"
+	slsa01 "github.com/in-toto/in-toto-golang/in_toto/slsa_provenance/v0.1"
+	dsselib "github.com/secure-systems-lab/go-securesystemslib/dsse"
+
+	serrors "github.com/slsa-framework/slsa-verifier/errors"
+	"github.com/slsa-framework/slsa-verifier/options"
+	"github.com/slsa-framework/slsa-verifier/verifiers/internal/gcb/keys"
+)
+
+var GCBBuilderIDs = []string{"https://cloudbuild.googleapis.com/GoogleHostedWorker@v0.2"}
+
+type v01IntotoStatement struct {
+	intoto.StatementHeader
+	Predicate slsa01.ProvenancePredicate `json:"predicate"`
+}
+
+type provenance struct {
+	Build struct {
+		// TODO: compare to verified provenance.
+		// IntotoStatement v01IntotoStatement `json:"intotoStatement"`
+	} `json:"build"`
+	Kind        string           `json:"kind"`
+	ResourceURI string           `json:"resourceUri"`
+	Envelope    dsselib.Envelope `json:"envelope"`
+}
+
+type gloudProvenance struct {
+	ImageSummary struct {
+		Digest               string `json:"digest"`
+		FullyQualifiedDigest string `json:"fully_qualified_digest"`
+		Registry             string `json:"registry"`
+		Repsitory            string `json:"repository"`
+	} `json:"image_summary"`
+	ProvenanceSummary struct {
+		Provenance []provenance `json:"provenance"`
+	} `json:"provenance_summary"`
+}
+
+type GCBProvenance struct {
+	gcloudProv                    *gloudProvenance
+	verifiedProvenance            *provenance
+	verifiedIntotoStatementStruct *v01IntotoStatement
```

Comment:
nit: Having struct in the name is a bit redundant
```suggestion
	verifiedIntotoStatement *v01IntotoStatement
```

Reply:
done, thanks.